mantidproject/mantid
# Tiled plot option for "Plot spectrum"

Enable the easy creation of a MultiLayer plot from right-clicking on a workspace.

At present, it is possible to create such a plot, but you have to plot each spectrum individually and then drag them onto a tiled plot, or write a Python script. This PR provides a way to do it from right-clicking a workspace in the GUI.

When used on a WorkspaceGroup, each workspace gets its own subplot, with the specified spectra in each. When used on a single workspace, each of the specified spectra is plotted in its own subplot.

**Example use case:**
After fitting several workspaces at once, the results are contained in a WorkspaceGroup. There is currently no easy way to visualise the results - you have to plot each fit individually in its own window, or have them all on the same axes (messy). With the new option, you can create a multiple plot of {data, fit, difference} for each workspace in its own subplot.

**To test:**
- Create a GroupWorkspace containing several multi-spectra MatrixWorkspaces.
- Right-click on the GroupWorkspace and choose "Plot Spectrum..."
- In the dialog, check the "Tiled Plot" option. Select a range of spectra to plot (or choose to plot all).
- (If there are a large number of workspaces in the group, it should warn you about this).
- Should get a plot of each workspace in its own subplot:
  - Axis titles should only be present on the leftmost column and bottom row (or second-to-last row if there is no plot underneath).
  - Axis scales should be the same for all subplots, and the scale should be large enough to show all the data in each plot. 
  - Font size of axis titles, labels and legends should be the same for all subplots.
  - There should not be excessive white space between the subplots.
  - Each subplot should have a legend to show which workspace it came from.
- Now right-click on one of the MatrixWorkspaces from the group, and do a tiled plot of this.
  - Each spectrum (from those selected) should be shown in its own subplot.

Fixes #9239.

[Release notes](https://github.com/mantidproject/mantid/blob/39583c9aa21c316b36e05de9d2ffbaffa8eac995/docs/source/release/v3.8.0/ui.rst#plotting-improvements) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
